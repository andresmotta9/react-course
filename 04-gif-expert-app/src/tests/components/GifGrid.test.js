import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from  "enzyme";
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Tests on GifGrid component', () => {

    const category = 'Goku';
    
    test('should <GifGrif /> match with the snapshot', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow( <GifGrid category={ category }/>);
        expect( wrapper  ).toMatchSnapshot();
    });
    
    test('should show the items when the images are loaded useFetchGifs', () => {
        const gifs = [
            {
            id: 'ABC',
            title: 'image title',
            url: 'https://localhost/image.jpg'
            },
            {
            id: 'ABCD',
            title: 'image title',
            url: 'https://localhost/image.jpg'
            },
        ]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow( <GifGrid category={ category }/>);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    })
    
})
