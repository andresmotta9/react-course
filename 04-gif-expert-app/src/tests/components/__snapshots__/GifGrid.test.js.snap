// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on GifGrid component should <GifGrif /> match with the snapshot 1`] = `
<Fragment>
  <h3>
     
    Goku
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Tests on GifGrid component should show the items when the images are loaded useFetchGifs 1`] = `
<Fragment>
  <h3>
     
    Goku
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="image title"
      url="https://localhost/image.jpg"
    />
    <GifGridItem
      id="ABCD"
      key="ABCD"
      title="image title"
      url="https://localhost/image.jpg"
    />
  </div>
</Fragment>
`;
